plugin/grpc_provider.go

```diff
@@ -119,7 +119,12 @@ func (p *GRPCProvider) GetSchema() (resp providers.GetSchemaResponse) {
 	resp.ResourceTypes = make(map[string]providers.Schema)
 	resp.DataSources = make(map[string]providers.Schema)
 
-	protoResp, err := p.client.GetSchema(p.ctx, new(proto.GetProviderSchema_Request))
+	// Some providers may generate quite large schemas, and the internal default
+	// grpc response size limit is 4MB. 64MB should cover most any use case, and
+	// if we get providers nearing that we may want to consider a finer-grained
+	// API to fetch individual resource schemas.
+	const maxRecvSize = 64 << 20
+	protoResp, err := p.client.GetSchema(p.ctx, new(proto.GetProviderSchema_Request), grpc.MaxRecvMsgSizeCallOption{MaxRecvMsgSize: maxRecvSize})
```

Comment:
Maybe we should mention in the comment above that this is an experimental setting, as an aid to the hypothetical future developer that upgrades the `grpc` dependency and suddenly sees a build error reported from this location.